pkg/networkpath/traceroute/runner.go

```diff
@@ -0,0 +1,238 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package traceroute
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/Datadog/dublin-traceroute/go/dublintraceroute/probes/probev4"
+	"github.com/Datadog/dublin-traceroute/go/dublintraceroute/results"
+
+	"github.com/google/uuid"
+)
+
+// TODO: are these good defaults?
+const (
+	DefaultSourcePort   = 12345
+	DefaultDestPort     = 33434
+	DefaultNumPaths     = 10
+	DefaultMinTTL       = 1
+	DefaultMaxTTL       = 30
+	DefaultDelay        = 50 //msec
+	DefaultReadTimeout  = 3 * time.Second
+	DefaultOutputFormat = "json"
+)
+
+// RunTraceroute wraps the implementation of traceroute
+// so it can be called from the different OS implementations
+func RunTraceroute(cfg Config) (NetworkPath, error) {
```

Comment:
Even if it's temporary, shall we clean up this code a bit more since it's going to be merged to `main` branch.

Also, maybe we can add a very explicit comment that this code is temporary and will be replaced by more definitive implementation?